jerry-core/ecma/base/ecma-gc.c

```diff
@@ -880,6 +907,39 @@ ecma_gc_free_native_pointer (ecma_property_t *property_p) /**< property */
   }
 } /* ecma_gc_free_native_pointer */
 
+/**
+ * Free specified arguments object.
+ */
+static size_t
+ecma_free_arguments_object (ecma_extended_object_t *ext_object_p) /**< arguments object */
+{
+  JERRY_ASSERT (ecma_get_object_type ((ecma_object_t *) ext_object_p) == ECMA_OBJECT_TYPE_PSEUDO_ARRAY);
+
+  size_t object_size = sizeof (ecma_unmapped_arguments_t);
+
+  if (ext_object_p->u.pseudo_array.extra_info & ECMA_ARGUMENTS_OBJECT_MAPPED)
+  {
+    ecma_mapped_arguments_t *mapped_arguments_p = (ecma_mapped_arguments_t *) ext_object_p;
+    object_size = sizeof (ecma_mapped_arguments_t);
+
+    ecma_compiled_code_t *byte_code_p = ECMA_GET_INTERNAL_VALUE_POINTER (ecma_compiled_code_t,
+                                                                         mapped_arguments_p->byte_code);
+
+    ecma_bytecode_deref (byte_code_p);
+  }
+
+  ecma_value_t *argv_p = (ecma_value_t *) (((uint8_t *) ext_object_p) + object_size);
+  ecma_unmapped_arguments_t *arguments_p = (ecma_unmapped_arguments_t *) ext_object_p;
+  uint32_t arguments_number = arguments_p->header.u.pseudo_array.u2.arguments_number;
+
+  for (uint32_t i = 0; i < arguments_number; i++)
+  {
+    ecma_free_value_if_not_object (argv_p[i]);
+  }
```

Comment:
What happens if new items are appended to the arguments object?

Reply:
These are freed by the object's property list.